src/app/core/services/cluster.service.spec.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {ClusterService} from './cluster';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
+import {TestBed} from '@angular/core/testing';
+import {MatDialogModule} from '@angular/material/dialog';
+import {AppConfigMockService} from '@test/services/app-config-mock';
+import {AppConfigService} from '@app/config.service';
+import {lastValueFrom} from 'rxjs';
+
+fdescribe('ClusterService', () => {
+  let service: ClusterService;
+  let httpController: HttpTestingController;
+
+  const mockProjectID = 'fake-project-id';
+  const mockClusterID = 'fake-cluster-id';
+  const mockMachineDeploymentID = 'fake-machine-deployment-id';
+  const mockRestRoot = 'mock-rest-root';
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [ClusterService, {provide: AppConfigService, useClass: AppConfigMockService}],
+      imports: [HttpClientTestingModule, MatDialogModule],
+    });
+    httpController = TestBed.inject(HttpTestingController);
+    service = TestBed.inject(ClusterService);
+  });
+
+  beforeEach(() => {
+    service['_newRestRoot'] = mockRestRoot;
+  });
+
+  afterEach(() => {
+    httpController.verify();
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should dispatch correct GET request when externalMachineDeploymentNodes() is called', () => {
```

Comment:
IMO, better to test like if individual machine deployment nodes should contain nodes of it's on machine based on `label` instead of checking checking if it is a `GET/POST` request.

If you have different opinion that this is test is important than please share your thoughts? It will be helpful.

Reply:
I'll add tests for that component too, but these tests are for the service